### PR TITLE
Graduate PodAffinityNamespaceSelector to GA

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Graduates PodAffinityNamespaceSelector to GA

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/enhancements/issues/2249

#### Does this PR introduce a user-facing change?

```release-note
pod-affinity namespace selector and cross-namespace quota graduated to GA. The feature gate PodAffinityNamespaceSelector is locked and will be removed in 1.26.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/2249-pod-affinity-namespace-selector
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/interpodaffinity/plugin.go [hunk 1]
```diff
@@ -85,15 +84,13 @@ func New(plArgs runtime.Object, h framework.Handle, fts feature.Features) (frame
...
-		enableNamespaceSelector: fts.EnablePodAffinityNamespaceSelector,
+		parallelizer: h.Parallelizer(),
+		args:         args,
+		sharedLister: h.SnapshotSharedLister(),
 	}
 
-	if pl.enableNamespaceSelector {
-		pl.nsLister = h.SharedInformerFactory().Core().V1().Namespaces().Lister()
-	}
+	pl.nsLister = h.SharedInformerFactory().Core().V1().Namespaces().Lister()
```
Comment (kerthcet) [inline pkg/scheduler/framework/plugins/interpodaffinity/plugin.go:92]: Combine them together?

**Comment by kerthcet** [comment]
looks good to me generally.

**Comment by msau42** [comment]
API changes lgtm

**Comment by msau42** [comment]
Also update the release note mentions that the feature gate is locked and will be removed in 1.26

**Comment by ahg-g** [comment]
> Also update the release note mentions that the feature gate is locked and will be removed in 1.26

Done, commit squashed as well

**Comment by liggitt** [comment]
/approve
for API changes

**Comment by ahg-g** [comment]
/assign @alculquicondor 

for scheduler lgtm

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Comment by ahg-g** [comment]
/hold cancel

can I get another lgtm pls

**Pull request merged**